### ISSUE TITLE
NVML Permissions

### ISSUE THREAD

**Issue by quasiben** [gpuopenanalytics/pynvml#19]
I'm filing this issue not for `pynvml` to resolve , but rather to raise awareness for permissioning issues around nvml to the pynvml devs. 

NVML can be administered in such away as to prevent non-admin/root users access to the library.  This would prevent nvml from querying performance counters.   For example:

```
(base) bzaitlen@dgx15:~$ nvidia-smi nvlink -g 0
GPU 0: Tesla V100-SXM2-32GB (UUID: GPU-96ab329d-7a1f-73a8-a9b7-18b4b2855f92)
NVML: Unable to get the NvLink link utilization counter control for link 0: Insufficient Permissions
```

I found this page on [admin pemissioning](https://developer.nvidia.com/nvidia-development-tools-solutions-ERR_NVGPUCTRPERM-permission-issue-performance-counters) which explains how to set/unset admin privileges.

Another way to easily check the status of permissions is to look for `RmProfilingAdminOnly` in driver params:

```
(base) bzaitlen@dgx15:~$ cat /proc/driver/nvidia/params | grep RmProfilingAdminOnly
RmProfilingAdminOnly: 1
```

**Comment by rjzamora** [comment]
Thaks @quasiben!  I do think it would be good for us to document these permissioning issues in pynvml - Perhaps some of the information you are sharing can be mentioned in the README

**Comment by rjzamora** [comment]
Thanks again @quasiben! I merged the doc changes in #20 -- Feel free to suggest other NVML-permissioning action items on the pynvml side.
